src/services/RetroGenerationService.ts

```diff
@@ -84,36 +87,36 @@ class RetroGenerationService {
                     defectsDetails =
                       defectsDetails +
                       " " +
-                      testType.defects[key].deficiencyRef +
+                      defectValue.deficiencyRef +
                       " (" +
-                      testType.defects[key].deficiencyCategory +
+                      defectValue.deficiencyCategory +
                       prsString +
-                      (testType.defects[key].additionalInformation.notes ? ", " + testType.defects[key].additionalInformation.notes : "") +
-                      (testType.defects[key].prohibitionIssued ? ", Prohibition was issued" : ", Prohibition was not issued") +
+                      (defectValue.additionalInformation.notes ? ", " + defectValue.additionalInformation.notes : "") +
+                      (defectValue.prohibitionIssued ? ", Prohibition was issued" : ", Prohibition was not issued") +
                       ")";
                   }
                   if (defectsDetails) {
                     defects = `Defects: ${defectsDetails};\r\n`;
                   }
-                  if (testType.reasonForAbandoning) {
-                    reasonForAbandoning = `Reason for abandoning: ${testType.reasonForAbandoning};\r\n`;
+                  if (testTypes[0].reasonForAbandoning) {
+                    reasonForAbandoning = `Reason for abandoning: ${testTypes[0].reasonForAbandoning};\r\n`;
                   }
-                  if (testType.additionalCommentsForAbandon) {
-                    additionalCommentsAbandon = `Additional comments for abandon: ${testType.additionalCommentsForAbandon};\r\n`;
+                  if (testTypes[0].additionalCommentsForAbandon) {
+                    additionalCommentsAbandon = `Additional comments for abandon: ${testTypes[0].additionalCommentsForAbandon};\r\n`;
                   }
-                  if (this.isPassingLECTestType(testType)) {
-                    LECNotes = "Modification type: " + testType.modType.code.toUpperCase() + "\r\n" + "Fuel type: " + testType.fuelType + "\r\n" + "Emission standards: " + testType.emissionStandard + "\r\n";
+                  if (this.isPassingLECTestType(testTypes)) {
```

Comment:
Can this isPassingLECTestType method be refactored to use the centralised list of LEC test type ids